### PR TITLE
Support INT block scale learning

### PR DESCRIPTION
**Pull request by realAsma**
### What does this PR do?

Type of change: new feature, new tests

Adds minimal INT block scale-learning support for the LAQ investigation branch:

- Converts eligible static INT block quantizers to `StaticBlockScaleQuantizer` after `max`, `mse`, or `local_hessian` scale initialization so LAQ can start from non-max initializers.
- Adds a fake dynamic INT block quantization path for weight-only dynamic max scale baselines with integer `num_bits` and `block_sizes: {"type": "dynamic"}`.
- Extends LAQ CPU unit coverage for INT3/INT2 static block max/mse initialization, frozen/original/dual/tied amax variants, and dynamic max weight-only forward.

### Usage

```python
quant_cfg = {
    "quant_cfg": [
        {"quantizer_name": "*", "enable": False},
        {
            "quantizer_name": "*weight_quantizer",
            "enable": True,
            "cfg": {"num_bits": 3, "block_sizes": {-1: 16, "type": "static"}},
        },
    ],
    "algorithm": {
        "method": "laq",
        "learnable_amax": ["pre", "post"],
        "tied_amax": False,
        "scale_algorithm": {"method": "mse"},
    },
}
```

Dynamic max scale baseline:

```python
quant_cfg = {
    "quant_cfg": [
        {"quantizer_name": "*", "enable": False},
        {
            "quantizer_name": "*weight_quantizer",
            "enable": True,
            "cfg": {"num_bits": 2, "block_sizes": {-1: 16, "type": "dynamic"}},
        },
    ],
    "algorithm": "max",
}
```

### QAD experiment plan

The QAD experiment plan is intentionally kept out of the PR description and
shared separately with the owner for review before any QAD jobs are launched.

### Testing

- `python_pwd -m ruff check modelopt/torch/quantization/tensor_quant.py modelopt/torch/quantization/nn/modules/tensor_quantizer.py tests/unit/torch/quantization/test_laq.py`
- `pytest_pwd tests/unit/torch/quantization/test_laq.py -q`
- `pre-commit run --files modelopt/torch/quantization/tensor_quant.py modelopt/torch/quantization/nn/modules/tensor_quantizer.py tests/unit/torch/quantization/test_laq.py`
- `pytest_pwd tests/unit/torch/quantization/test_laq.py -q` after pre-commit

### Before your PR is "Ready for review"

- Is this change backward compatible?: Yes
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: N/A
- Did you write any new necessary tests?: Yes
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: N/A
- Did you get Claude approval on this PR?: N/A, draft PR for experiment-plan review

### Additional Information

Draft PR targeting the LAQ branch for owner review before launching QAD jobs.

### CODE REVIEW

**Comment by realAsma** [comment]
Keep tied Dual LSQ as a second-pass reducer if independent Dual LSQ is unstable or too expensive.

> what does this mean? Not clear to me.

**Comment by realAsma** [comment]
Train CE loss.
Eval CE loss.
Eval KL/KD loss when teacher logits are available. -> fix QAD uses KL div training loss.

**Comment by realAsma** [comment]
Export compatibility status for static INT block variants.

> Ignore this please

File: modelopt/torch/quantization/nn/modules/tensor_quantizer.py [hunk 1]
```diff
@@ -817,17 +842,21 @@ def _fake_quantize(self, inputs):
...
-                self._get_bias(inputs),
-                self._num_bits,
-                self.block_sizes.get("scale_bits", None),
-                getattr(self, "_trt_high_precision_dtype", None),
-                getattr(self, "_onnx_quantizer_type", None),
-                self._pass_through_bwd,
-            )
+            bias = self._get_bias(inputs)
+            if isinstance(self._num_bits, int) and self.block_sizes.get("scale_bits") is None:
+                outputs = self._fake_dynamic_int_block_quantize(inputs, block_size, bias)
```
Comment (realAsma) [inline modelopt/torch/quantization/nn/modules/tensor_quantizer.py]: Can we move this to tensor_quant.py when dynamic_block_quant is called?

Comment (realAsma) [reply modelopt/torch/quantization/nn/modules/tensor_quantizer.py]: > _:robot: Bot comment._

Fixed in `ec72e810b2`: moved the dynamic INT block fake-quant helper into `tensor_quant.py` as `dynamic_int_block_quant`, with `TensorQuantizer` only dispatching to it.

**Comment by realAsma** [comment]
> _:robot: Bot comment._

Addressed the QAD-plan comments in the PR body:

- Removed the unclear tied-Dual-LSQ fallback wording from the first grid.
- Updated metrics so QAD training KL/KD loss is the primary training objective, with train CE only called out if logged separately.
- Removed export compatibility from the reviewed metrics list as requested.

Plan comments addressed: https://github.com/NVIDIA/Model-Optimizer/pull/1795#issuecomment-4774087836, https://github.com/NVIDIA/Model-Optimizer/pull/1795#issuecomment-4774093497, https://github.com/NVIDIA/Model-Optimizer/pull/1795#issuecomment-4774096187
